src/app/components/course-island/course-island.component.html

```diff
@@ -18,17 +18,27 @@ <h2 class="tui-island__title">{{ course.name }}</h2>
         ></tui-tag>
         <div class="tui-form__buttons">
             <a
-                *ngIf="hasViewPermission"
+                *ngIf = "user.is_teacher"
+                appearance="primary"
                 [routerLink]="['/course', course.id]"
+                class="tui-form__button"
+                size="m"
+                tuiButton
+            >
+                View
+            </a>
+            <a
+                *ngIf="hasViewPermission && endDate > currentDate"
```

Comment:
You should add `!user.is_teacher` as it might show 2 buttons if the user is a teacher and this if is true.

Reply:
actually you can have `user.is_student`

Reply:
Okay got it